File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/dto/experiment/id/ExperimentIdentifier.js
/**
 * Experiment identifier.
 * 
 * @author pkupczyk
 */
define([ "stjs", "dto/common/id/ObjectIdentifier", "dto/experiment/id/IExperimentId" ], function(stjs, ObjectIdentifier, IExperimentId) {
	/**
	 * @param identifier
	 *            Experiment identifier, e.g.
	 *            "/MY_SPACE/MY_PROJECT/MY_EXPERIMENT".
	 */
	var ExperimentIdentifier = function(identifier) {
		ObjectIdentifier.call(this, identifier);
	};
	stjs.extend(ExperimentIdentifier, ObjectIdentifier, [ ObjectIdentifier, IExperimentId ], function(constructor, prototype) {
		prototype['@type'] = 'dto.experiment.id.ExperimentIdentifier';
		constructor.serialVersionUID = 1;
	}, {});
	return ExperimentIdentifier;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/as/api/v3/dto/experiment/id/ExperimentIdentifier.java
/*
 * Copyright 2012 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.as.api.v3.dto.experiment.id;

import ch.ethz.sis.openbis.generic.as.api.v3.dto.common.id.ObjectIdentifier;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * Experiment identifier.
 * 
 * @author pkupczyk
 */
@JsonObject("dto.experiment.id.ExperimentIdentifier")
public class ExperimentIdentifier extends ObjectIdentifier implements IExperimentId
{

    private static final long serialVersionUID = 1L;

    /**
     * @param identifier Experiment identifier, e.g. "/MY_SPACE/MY_PROJECT/MY_EXPERIMENT".
     */
    public ExperimentIdentifier(String identifier)
    {
        super(identifier);
    }

    public ExperimentIdentifier(String spaceCode, String projectCode, String experimentCode)
    {
        this("/" + spaceCode + "/" + projectCode + "/" + experimentCode);
    }

    //
    // JSON-RPC
    //

    @SuppressWarnings("unused")
    private ExperimentIdentifier()
    {
        super();
    }

}
